packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -336,6 +350,8 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     this.#isUnlocked = isUnlocked;
 
     this.#accountsAPI.isAccountsAPIEnabled = useAccountsAPI;
+    this.#useTokenDetection = useTokenDetection();
+    this.#useExternalServices = useExternalServices();
```

Comment:
Same here, lets save the functions instead of saving the result from the controller init. That way we can get up-to-date values after the controller initialisation.